sound/soc/sof/ipc4-pcm.c

```diff
@@ -504,7 +504,7 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 		snd_mask_set_format(fmt, SNDRV_PCM_FORMAT_S32_LE);
 	}
 
-	rate->min = ipc4_copier->available_fmt.sink_audio_fmts->sampling_frequency;
+	rate->min = ipc4_copier->available_fmt.sink_audio_fmts->audio_fmt.sampling_frequency;
```

Comment:
we need to add a TODO where this rate->min is updated by parsing all the formats instead of taking the first element.

Reply:
@plbossart A start is here -> https://github.com/thesofproject/linux/pull/4137/commits/641f51461e2c433d40fbcbd6fd4afa76d78c154a
This enumerate all formats and picks one if there's a match with FE (i.e. align BE to FE rate if supported), but picks the first one if no match (assuming topology will have a SRC to handle the difference).

I couldn't really find any examples in ASoC where mix/max is set to a range in dai fixup.